packages/plugin-link/src/Link.ts

```diff
@@ -145,11 +145,18 @@ export class Link<T extends JWPluginConfig = JWPluginConfig> extends JWPlugin<T>
         if (params.target) {
             link.modifiers.get(Attributes).set('target', params.target);
         }
-        return params.context.execCommand<Char>('insertText', {
-            text: label || link.url,
-            formats: new Modifiers(link),
-            context: params.context,
-        });
+        return this.editor.execWithRange<Char>(
+            [
+                [range.start, RelativePosition.AFTER],
+                [range.end, RelativePosition.BEFORE],
+            ],
```

Comment:
You probably want to use `VRange.clone(range)` in this case. However, this is going to diverge from gspec. Wouldn't it be better to clone the range from Odoo side and call `execWithRange` on the previously cloned range to do the additional stuff you would like to do with the "previous" range ?